crates/namada/src/ledger/vp_host_fns.rs

```diff
@@ -188,7 +188,9 @@ where
             Ok(false)
         }
         Some(&write_log::StorageModification::InitAccount { .. }) => Ok(true),
-        Some(&write_log::StorageModification::Temp { .. }) => Ok(true),
+        Some(&write_log::StorageModification::Temp { .. }) => {
```

Comment:
The API we provide to the VPs is different to the one we have in protocol or expose to transactions. For VPs we report an error whereas in protocol and for transactions we simply ignore temporary writes and try to look for other storage modifications or directly from storage. I tend to like this latter but I can understand that returning an error might be more explicit. Should we try to unify the api and apply the same logic everywhere (either errors or read-through?)

Reply:
I agree that the latter is better and I'm for unifying this in VPs. If a VP allows temp values, it has to read them explicitly so ignoring them for non-temp reads is fine